lib/ui5Framework/maven/Registry.js

```diff
@@ -0,0 +1,108 @@
+import {getLogger} from "@ui5/logger";
+import fetch from "make-fetch-happen";
+import xml2js from "xml2js";
+import {promisify} from "node:util";
+import {pipeline} from "node:stream/promises";
+import fs from "graceful-fs";
+const log = getLogger("ui5Framework:maven:Registry");
+
+class Registry {
+	/**
+	 * @param {object} parameters Parameters
+	 * @param {string} parameters.endpointUrl Maven's endpoint URL
+	 */
+	constructor({endpointUrl}) {
+		if (!endpointUrl) {
+			throw new Error(`Registry: Missing parameter "endpointUrl"`);
+		}
+		this._endpointUrl = endpointUrl;
+		if (!this._endpointUrl.endsWith("/")) {
+			this._endpointUrl += "/";
+		}
```

Comment:
For error analysis purposes, we should add a verbose log for the effective endpointUrl value

Reply:
Done